canonical-web-and-design/maas-ui
# refactor(ui): license keys slice

## Done

- Migrate the license keys actions and reducers to a slice.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Create a windows image: https://github.com/canonical-web-and-design/maas-ui/blob/master/HACKING.md#creating-a-fake-windows-image.
- Go to Settings -> License keys.
- Create, edit and delete license keys (can just use `XXXXX-XXXXX-XXXXX-XXXXX-XXXXX` for a key). All actions should work.

## Fixes

Fixes: #1409.
